src/pages/en/guides/data-fetching.md

```diff
@@ -4,51 +4,125 @@ title: Data Fetching
+const randomUser = data.results[0]
+---
+<!-- Data fetched at build can be rendered in HTML -->
+<h1>User</h1>
+<h2>{randomUser.name.first} {randomUser.name.last}</h2>
+
+<!-- Data fetched at build can be passed to components as props -->
+<Contact client:load email={randomUser.email} />
+<Location city={randomUser.location.city} />
+```
+
+### requestOptions
+
+You can specify any necessary request options for your API fetch call in your Astro page component script.
 
 ```astro
 ---
-// Movies.astro
-const response = await fetch('https://example.com/movies.json');
+import MyReactApp from '../components/MyReactApp.jsx';
+
+const headers = new Headers()
+headers.append("API_KEY", import.meta.env.API_KEY)
+
+const requestOptions = {
+  method: "GET",
+  headers: headers,
+  redirect: "follow",
+  cors:"no-cors"
+};
+
+const response = await fetch(
+  `https://api.ebird.org/v2/data/obs/CA-PE/recent/notable?detail=full`,
+  requestOptions
+);
 const data = await response.json();
-// Remember: Astro component scripts log to the CLI
 console.log(data);
 ---
-<!-- Output the result to the page -->
-<div>{JSON.stringify(data)}</div>
+  <MyReactApp dataForApp={data}/>
 ```
 
-## Top-level await
+### GraphQL queries
 
-`await` is another native JavaScript feature that lets you await the response of some asynchronous promise ([MDN<span class="sr-only">- await</span>](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await)). Astro supports `await` in the top-level of your component script.
+Astro can also use `fetch()` to query a GraphQL server at build time with any valid GraphQL query. 
 
-**Important:** These are not yet available inside of non-page Astro components. Instead, do all of your data loading inside of your pages, and then pass them to your components as props.
+```astro
+---
+import BaseLayout from '../../layouts/BaseLayout.astro';
+
+const response = await fetch("https://graphql-weather-api.herokuapp.com", 
+  {
+    method:'POST',
+    headers: {'Content-Type':'application/json'},
+    body: JSON.stringify({
+      query: `
+        query getWeather($name:String!) {
+            getCityByName(name: $name){
+              name
+              country
+              weather {
+                summary {
+                    description
+                }
+              }
+            }
+          }
+        `,
+      variables: {
+          name: "Toronto",
+      },
+    }),
+  })
+
+const json = await response.json();
+const weather = json.data
+---
+<BaseLayout title = "GraphQL API fetch in Astro" >
+  <h1>Fetching Weather at build time</h2>
+  <h2>{weather.getCityByName.name}, {weather.getCityByName.country}</h3>
+  <p>Weather: {weather.getCityByName.weather.summary.description}</p>
+</BaseLayout>
+```
+> 💡 Remember, all data in Astro components is **fetched once, at build time**. For data that can be re-fetched multiple times client side, use a [framework component](/en/core-concepts/framework-components).
```

Comment:
If you're calling this out, it may make sense to mention the fact that fetch() will be called every time you render a component. So this may mean once during build time, but it may mean on every refresh during dev.

For what it's worth, this is something that we want to fix before v1.0.0 with better caching rules. But for now, that's the reality.